x/cash/dynamicfee.go

```diff
@@ -42,98 +42,63 @@ import (
 
 type DynamicFeeDecorator struct {
 	auth x.Authenticator
-	ctrl CoinMover
+	ctrl Controller
```

Comment:
q: why not use the `CoinMover` interface?

Reply:
I can be used again once we don't need the `Balance` method.

Reply:
We can leave as is for this PR.
In general, I would say we can make three interfaces:

CoinMover - read only
CoinMover + MoveCoins - user privledge
CoinMover + MoveCoins + IssueCoins  - "admin" privledge

And then we look very closely at any extension that takes the IssueCoins interface